### PR TITLE
Make router metal-ipi jobs optional again

### PR DESCRIPTION
**Pull request by sgreene570**
Also, only run the metal ipv6 job on every PR.

This is a follow up to https://github.com/openshift/release/pull/12570, which clobbered the existing `always_run` and `optional` flags for the router metal-ipi tests.

### CODE REVIEW

File: ci-operator/jobs/openshift/router/openshift-router-master-presubmits.yaml [hunk 1]
```diff
@@ -66,7 +66,7 @@ presubmits:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e,?($|\s.*)
   - agent: kubernetes
-    always_run: true
+    always_run: false
```
Comment (Miciah) [inline ci-operator/jobs/openshift/router/openshift-router-master-presubmits.yaml:69]: Is this change intentional?  We had `always_run: true` even before #12570.

**Review by Miciah — COMMENTED** [review]
Do we need both `ci/prow/e2e-metal-ipi-ovn-dualstack` and `ci/prow/e2e-metal-ipi-ovn-router`?  Should the latter be renamed to `ci/prow/e2e-metal-ipi-ovn-ipv6`?

File: ci-operator/jobs/openshift/router/openshift-router-master-presubmits.yaml (hunk 1 above)
Comment (sgreene570) [reply ci-operator/jobs/openshift/router/openshift-router-master-presubmits.yaml:69]: `always_run: true` was set for `pull-ci-openshift-router-master-e2e-metal-ipi`, which has been renamed to ` pull-ci-openshift-router-master-e2e-metal-ipi-ovn-ipv6`. This diff sets `always_run: false` for the new dualstack job (the prior metal ipi job was v6, not dualstack, so I figured we don't want the new dualstack job to always run).

**Comment by sgreene570** [comment]
> Do we need both `ci/prow/e2e-metal-ipi-ovn-dualstack` and `ci/prow/e2e-metal-ipi-ovn-router`? Should the latter be renamed to `ci/prow/e2e-metal-ipi-ovn-ipv6`?

`ci/prow/e2e-metal-ipi-ovn-router` is the same job as `ci/prow/e2e-metal-ipi-ovn-dualstack`, just with the router e2e tests added in (by default the metal-ipi jobs only test a subset of the tests available in origin). Should we rename `ci/prow/e2e-metal-ipi-own-router` to `ci/prow/e2e-metal-ipi-ovn-dualstack-router-tests`, or something along those lines? I believe the intention was to remove the `ci/prow/e2e-metal-ipi-ovn-router` job when the base metal-ipi job ultimately includes more of the standard e2e tests.

**Comment by Miciah** [comment]
> > Do we need both `ci/prow/e2e-metal-ipi-ovn-dualstack` and `ci/prow/e2e-metal-ipi-ovn-router`? Should the latter be renamed to `ci/prow/e2e-metal-ipi-ovn-ipv6`?
> 
> `ci/prow/e2e-metal-ipi-ovn-router` is the same job as `ci/prow/e2e-metal-ipi-ovn-dualstack`, just with the router e2e tests added in (by default the metal-ipi jobs only test a subset of the tests available in origin). Should we rename `ci/prow/e2e-metal-ipi-ovn-router` to `ci/prow/e2e-metal-ipi-ovn-dualstack-router-tests`, or something along those lines? I believe the intention was to remove the `ci/prow/e2e-metal-ipi-ovn-router` job when the base metal-ipi job ultimately includes more of the standard e2e tests, so im fine leaving the name alone for now (but could be convinced to update it as well).

Thanks for explaining!  I'm fine with leaving the name as is.

/lgtm
